search-commons/src/main/java/no/unit/nva/search2/OpenSearchAwsClient.java

```diff
@@ -1,124 +1,142 @@
         var stringQueryBuilder = QueryBuilders.queryStringQuery(luceneParameters);
         stringQueryBuilder.defaultOperator(Operator.AND);
         var fields = query.removeValue(FIELDS);
         if (nonNull(fields)) {
-            Arrays.stream(fields.split(",")).forEach(stringQueryBuilder::field);
+            Arrays.stream(fields.split(COMMA)).forEach(stringQueryBuilder::field);
         }
-        return Stream.of(new Tuple<>(stringQueryBuilder, query));
+        return Stream.of(new QueryBuilder(stringQueryBuilder, query, mediaType));
     }
 
-    @Override
-    public void close() throws IOException {
-        client.close();
-    }
-
-    private SearchSourceBuilder searchSourceWithAggregation(Tuple<QueryStringQueryBuilder, ResourceAwsQuery> tuple) {
-        var builder = new SearchSourceBuilder().query(tuple.v1());
-        var searchAfter = tuple.v2().removeValue(SEARCH_AFTER);
+    private QueryBuilderSource searchSourceWithAggregation(QueryBuilder queryBuilder) {
+        var builder = new SearchSourceBuilder().query(queryBuilder.buider());
+        var query = queryBuilder.query();
+        var searchAfter = query.removeValue(SEARCH_AFTER);
         if (nonNull(searchAfter)) {
-            var sortKeys = searchAfter.split(",");
+            var sortKeys = searchAfter.split(COMMA);
             builder.searchAfter(sortKeys);
         }
+
         RESOURCES_AGGREGATIONS.forEach(builder::aggregation);
-        return builder;
+
+        builder.size(query.getValue(SIZE).as());
+        builder.from(query.getValue(FROM).as());
+        Arrays.stream(query.getValue(SORT).<String>as().split(COMMA))
+            .map(sort -> sort.split(COLON))
+            .map(this::expandSortKeys)
+            .forEach(params -> builder.sort(params.v1(), params.v2()));
+
+        return new QueryBuilderSource(builder, query.openSearchUri(), queryBuilder.mediaType());
     }
 
-    private SearchRequest searchRequest(SearchSourceBuilder sourceBuilder) {
-        return new SearchRequest(RESOURCES).source(sourceBuilder);
+    private Tuple<String, SortOrder> expandSortKeys(String... strings) {
+        var sortOrder = strings.length == 2 ? SortOrder.fromString(strings[1]) : SortOrder.ASC;
+        var luceneKey = SortKeys.keyFromString(strings[0]).getLuceneField();
+        return new Tuple<>(luceneKey, sortOrder);
+    }
+
+    @JacocoGenerated
+    private HttpRequest httpRequest(QueryBuilderSource qbs) {
+        logger.info(qbs.requestUri().toString());
+        return HttpRequest
+                   .newBuilder(qbs.requestUri())
+                   .headers(ACCEPT, qbs.mediaType(), AUTHORIZATION_HEADER, jwtProvider.getValue().getToken())
+                   .POST(HttpRequest.BodyPublishers.ofString(qbs.source().toString())).build();
+    }
+
+    private OpenSearchSwsResponse sendHttpRequest(HttpRequest httpRequest) {
+        return
+            attempt(() -> httpClient.send(httpRequest, bodyHandler))
+                .map(this::handleResponse)
```

Comment:
a response in a method called request is not very intuitive, but not a big issue